x/emissions/module/rewards_v2_test.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/allora-network/allora-chain/x/emissions/types"
+)
+
+func (s *ModuleTestSuite) TestGetWorkerScoreInferenceTask() {
+	timeNow := uint64(time.Now().UTC().Unix())
+
+	// Create a topic
+	topicIds, err := mockCreateTopics(s, 1)
+	s.NoError(err, "Error creating topic")
+	topicId := topicIds[0]
+
+	// Create and register 2 reputers in topic
+	reputers, err := mockSomeReputers(s, topicId)
+	s.NoError(err, "Error creating reputers")
+
+	// Create and register 2 workers in topic
+	workers, err := mockSomeWorkers(s, topicId)
+	s.NoError(err, "Error creating workers")
+
+	// Add a lossBundle for each reputer
+	var reputersLossBundles []*types.LossBundle
+	reputer1LossBundle := types.LossBundle{
+		TopicId: topicId,
+		Reputer: reputers[0].String(),
+		InfererLosses: []*types.WorkerAttributedLoss{
+			{
+				Worker: workers[0].String(),
+				Value: cosmosMath.NewUint(90),
+			},
+			{
+				Worker: workers[1].String(),
+				Value: cosmosMath.NewUint(100),
+			},
+		},
+		// Increased loss when removing for worker 1
+		OneOutLosses: []*types.WorkerAttributedLoss{
+			{
+				Worker: workers[0].String(),
+				Value: cosmosMath.NewUint(115),
+			},
+			{
+				Worker: workers[1].String(),
+				Value: cosmosMath.NewUint(100),
+			},
+		},
+	}
+	reputersLossBundles = append(reputersLossBundles, &reputer1LossBundle)
+	reputer2LossBundle := types.LossBundle{
+		TopicId: topicId,
+		Reputer: reputers[1].String(),
+		InfererLosses: []*types.WorkerAttributedLoss{
+			{
+				Worker: workers[0].String(),
+				Value: cosmosMath.NewUint(85),
+			},
+			{
+				Worker: workers[1].String(),
+				Value: cosmosMath.NewUint(100),
+			},
+		},
+		// Increased loss when removing for worker 1
+		OneOutLosses: []*types.WorkerAttributedLoss{
+			{
+				Worker: workers[0].String(),
+				Value: cosmosMath.NewUint(120),
+			},
+			{
+				Worker: workers[1].String(),
+				Value: cosmosMath.NewUint(100),
+			},
+		},
+	}
+	reputersLossBundles = append(reputersLossBundles, &reputer2LossBundle)
+
+	err = s.emissionsKeeper.InsertLossBudles(s.ctx, topicId, timeNow, types.LossBundles{LossBundles: reputersLossBundles})
+	s.NoError(err, "Error adding lossBundle for worker")
+
+	// Get LossBundles
+	lossBundles, err := s.emissionsKeeper.GetLossBundles(s.ctx, topicId, timeNow)
+	s.NoError(err, "Error getting lossBundles")
+
+	// Get reputers stakes and reported losses for each worker
+	var reputersStakes []float64
+	var reputersWorker1ReportedLosses []float64
+	var reputersWorker2ReportedLosses []float64
+	var reputersWorker1ReportedOneOutLosses []float64
+	var reputersWorker2ReportedOneOutLosses []float64
+
+	for _, lossBundle := range lossBundles.LossBundles {
+		reputerAddr, err := sdk.AccAddressFromBech32(lossBundle.Reputer)
+		s.NoError(err, "Error getting reputerAddr")
+
+		reputerStake, err := s.emissionsKeeper.GetStakeOnTopicFromReputer(s.ctx, topicId, reputerAddr)
+		s.NoError(err, "Error getting reputerStake")
+
+		reputerStakeFloat, _ := reputerStake.BigInt().Float64()
```

Comment:
so this is where we need to be weary about truncation vs the correct encoding in float because if the value is large, it'll just get rounded to the closest value that the float can represent or +Inf